### ISSUE TITLE
Autosubmit must never re-use an expid

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1072]
( This is a template for bug reports. Please fill in the relevant information and delete the rest. )

Hello @dbeltrankyl and @kinow, (cc @mcastril )

### Autosubmit Version

Any.

### Expid affected(If applicable)

NA

### Which task has issues? Where is the log(If applicable)

NA

### Summary

After you delete an experiment with `autosubmit delete`, it frees up that experiment's current expid. If that expid is the latest expid available, then the next command will re-use the same expid.

### Steps to reproduce

1. `autosubmit expid ....`
2. No other experiments created
3. `autosubmit delete $expid`
4. `autosubmit expid ....`

Now you have the same $expid.

### What is the current bug behavior?

An experiment ID is used twice.

### What is the expected correct behavior?

Experiment ID's must be used to uniquely identify experiments. If the deleted experiment had been used to upload data to an archival system, then the experiment itself was archived and then deleted, it is expected it could be unarchived or, at least, that no other experiment would use the same expid (risking to re-upload data to the same archival system under the same expid).

### Relevant logs and/or screenshots(if applicable)

NA

### Any other relevant information(if applicable)

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 30, 2023, 11:19

I can explain the reason why this behavior has been maintained, but the reason could not sound flawless.

Autosubmit would only re-use an ID if when this is deleted this is the last one in the list.

With the degree of Autosubmit usage in the department, we assume that by the time someone has deleted an experiment, if no one else has created a newer experiment, this experiment was a mistake or was used for nothing.

This is somehow risky but we were more concerned about exhausting the IDs.

But it's true that this scenario can not be the same in other instances... so it's a good point to not re-using them.

**Comment by dbeltrankyl** [comment]
Hello @kinow 

Should we add this for 4.2.0?

**Comment by kinow** [comment]
I think this is part or related to @ntorqulu 's recent work on traceability. I think instead of removing entries, that PR will update it to DELETED (PR #2980 linked above). I'm nearly done reviewing that one, just pending two files. But Núria had to stop due to the merge of my PR to remove autosubmit.py, and due to other issues she had to work on. I guess it should still be good for 4.2.0, unless we are in a hurry to release that version.

**Comment by ntorqulu** [comment]
Correct, we'll keep the experiment entries as tombstones  in the db so that new expid's cannot be repeated. This way we guarantee that expid's are not reused.
